feat(roles-access): Permissions grid for organization settings role access

liyiy opened the pull request:
## Problem
The current implementation of permission settings display was not super intuitive. Having it in a grid format makes it easier to view permissions for all roles and to update them accordingly.
<!-- Who are we building for, what are their needs, why is this important? -->

## Changes
<img width="1140" alt="Screen Shot 2022-12-14 at 1 27 24 AM" src="https://user-images.githubusercontent.com/25164963/207523191-b829d3ac-d580-4ef9-85ba-5dc30b715948.png">
<img width="673" alt="Screen Shot 2022-12-14 at 1 27 38 AM" src="https://user-images.githubusercontent.com/25164963/207523193-af14e520-105b-4374-a46a-79b455adeeec.png">
<img width="1107" alt="Screen Shot 2022-12-14 at 1 29 52 AM" src="https://user-images.githubusercontent.com/25164963/207523224-191a8479-5abb-40fe-b709-66355a7d67c6.png">
<img width="452" alt="Screen Shot 2022-12-14 at 2 06 26 AM" src="https://user-images.githubusercontent.com/25164963/207528742-13c30d26-2e8a-496c-8b59-7897e24f14b2.png">
<!-- If there are frontend changes, please include screenshots. -->
<!-- If a reference design was involved, include a link to the relevant Figma frame! -->

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## How did you test this code?

<!-- Briefly describe the steps you took. -->
<!-- Include automated tests if possible, otherwise describe the manual testing routine. -->

liyiy committed:
add permissions grid

liyiy committed:
types and lemon toast to indicate successful update

liyiy committed:
be able to delete roles on the role modal

liyiy committed:
fix buttons and more detailed lemon toast update

liyiy committed:
Merge branch 'master' into rba-permissions-grid

liyiy committed:
tooltip help for view access

liyiy commented on frontend/src/scenes/organization/Settings/Permissions/permissionsLogic.tsx:
```diff
@@ -50,6 +67,28 @@ export const permissionsLogic = kea<permissionsLogicType>([
+                })
+                if (updatedRole) {
+                    actions.updateRole(updatedRole)
+                    lemonToast.success(`${role.name} role ${resourceType} edit access updated`)
```
Opted for a toast display here indicating a successful edit access update instead of a save button since updating these organizational settings is instantaneous (toast) vs in a form (save btn). This is similar to how we have `X member has been invited` or `Your configuration has been saved` toasts in our current organization settings page when we invite a member or edit an organization setting.

liyiy replied:
yepp

annikaschmid commented:
For the "create role" popup, I would replace the permissions dropdown with checkboxes as well:

View Feature Flags [. ]
Edit Feature Flags [. ]

Or the popup doesn't have permissions at all, and you only create the role and then modify permissions in the table. Because once we have more permissions, the popup would get quite big

annikaschmid commented:
Do you want to stick to calling this tap "role access" or rename it to e.g. Role-based access or Feature Permissions?

neilkakkar commented:
Should the "view" checkbox be disabled given that they can't change it anyway?

liyiy commented:
> @liyiy Could we show the check boxes for "view" as disabled? I rather have a visual difference here from the "edit" column, rather than the tooltip on hover.

They are disabled

liyiy committed:
update tab name and remove permissions editing from role modal

liyiy commented:
@annikaschmid 

> Do you want to stick to calling this tap "role access" or rename it to e.g. Role-based access or Feature Permissions?

Updated to "Role-based access"

<img width="643" alt="Screen Shot 2022-12-15 at 12 55 44 PM" src="https://user-images.githubusercontent.com/25164963/207933171-393a628c-756b-4870-842a-aca72ffc36b0.png">

> Or the popup doesn't have permissions at all, and you only create the role and then modify permissions in the table. Because once we have more permissions, the popup would get quite big

Removed permissions setting from the create role modal :) 

<img width="589" alt="Screen Shot 2022-12-15 at 12 55 31 PM" src="https://user-images.githubusercontent.com/25164963/207933114-79052879-9cb0-4db4-b655-7d5b8b6206b9.png">

annikaschmid commented:
> > @liyiy Could we show the check boxes for "view" as disabled? I rather have a visual difference here from the "edit" column, rather than the tooltip on hover.
> 
> They are disabled. Lemon checkbox just doesn't differentiate disabled checkboxes with a gray color so I added the tooltip

I wonder if it's worth extending lemon-ui in this case? Do we have a decision making framework for stuff like this?

liyiy commented:
> I wonder if it's worth extending lemon-ui in this case? Do we have a decision making framework for stuff like this?

Colors for checkboxes were deprecated according to https://github.com/PostHog/posthog/pull/9357#pullrequestreview-933783868, it makes sense to me that disabled checkboxes could be more grayed out but I'll confirm with the head lemon himself @Twixes 

@annikaschmid Do you think that's very blocking for this PR?

liyiy committed:
gray checkboxes for disabled options in grid
